### PR TITLE
Add third option (Friday Institute) to virtual dropdown when creating/updating workshops

### PR DESCRIPTION
**Pull request by bencodeorg**
Adds a third option to the "Is this a virtual workshop?" dropdown when creating or updating a workshop.

Implemented as a new attribute on the workshop model, `third_party_provider`. There's some hairy logic in the `workshop_form` JSX file to make this work in a single dropdown, would appreciate feedback on how to make it cleaner. 

## Testing story

Tested creating and editing workshops via UI locally and verified data being stored in database, as well as adding a couple of unit tests to limit new model attributes.

## Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -78,7 +86,7 @@ export class WorkshopForm extends React.Component {
       id: PropTypes.number.isRequired,
       facilitators: PropTypes.array.isRequired,
       location_name: PropTypes.string.isRequired,
-      location_address: PropTypes.string.isRequired,
+      location_address: PropTypes.string,
```
Comment (bencodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:89]: Fixes unrelated console error -- oversight in my [previous PR making address optional](https://github.com/code-dot-org/code-dot-org/pull/35031).

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 2]
```diff
@@ -737,13 +748,33 @@ export class WorkshopForm extends React.Component {
     return value;
   };
 
+  currentVirtualStatus = () => {
```
Comment (bencodeorg) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:751]: I don't love this in particular, if anyone has suggestions.

Comment (islemaster) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:751]: This seems correct.  I know it's a bit complicated, but hopefully we don't have to preserve this logic forever.

**Review by islemaster — APPROVED** [review]

**Review by molly-moen — APPROVED** [review]
Nice!

**Pull request merged**